### PR TITLE
TRUNK-5837: Cohort Domain - Switching from Hibernate Mappings to Annotations

### PR DESCRIPTION
**Pull request by alinmihaila99**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5837

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/Cohort.java [hunk 1]
```diff
@@ -21,16 +34,28 @@
 /**
  * This class represents a list of patientIds.
  */
+@Entity
+@Table(name = "cohort")
+@AttributeOverride(name = "uuid", column = @Column(name = "uuid", length = 38, unique = true))
 public class Cohort extends BaseChangeableOpenmrsData {
-	
+
```
Comment (HerbertYiga) [inline api/src/main/java/org/openmrs/Cohort.java:41]: @alinmihaila99  thanks though this came out with a few unwanted code formats

File: api/src/main/java/org/openmrs/Cohort.java [hunk 2]
```diff
@@ -21,16 +34,28 @@
...
+	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "cohort_id_gen")
+	@SequenceGenerator(name = "cohort_id_gen", sequenceName = "cohort_cohort_id_seq")
 	private Integer cohortId;
-	
+
+	@Column(nullable = false)
 	private String name;
-	
+
+	@Column(nullable = false, length = 1000)
```
Comment (sherrif10) [inline api/src/main/java/org/openmrs/Cohort.java:53]: Why did you add the length when you had specified as null

Comment (alinmihaila99) [reply api/src/main/java/org/openmrs/Cohort.java:53]: The field can't be null

**Review by Ruhanga — CHANGES_REQUESTED** [review]
@alinmihaila99, could you resolve the merge conflicts?

**Review by Ruhanga — APPROVED** [review]
